src/main/kotlin/no/nav/familie/ks/sak/config/KafkaConfig.kt

```diff
@@ -0,0 +1,97 @@
+package no.nav.familie.ks.sak.config
+
+import no.nav.familie.kontrakter.felles.Applikasjon
+import org.apache.kafka.clients.CommonClientConfigs
+import org.apache.kafka.clients.consumer.ConsumerConfig
+import org.apache.kafka.clients.producer.ProducerConfig
+import org.apache.kafka.common.config.SslConfigs
+import org.apache.kafka.common.serialization.StringDeserializer
+import org.apache.kafka.common.serialization.StringSerializer
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Profile
+import org.springframework.kafka.annotation.EnableKafka
+import org.springframework.kafka.config.ConcurrentKafkaListenerContainerFactory
+import org.springframework.kafka.core.ConsumerFactory
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory
+import org.springframework.kafka.core.DefaultKafkaProducerFactory
+import org.springframework.kafka.core.KafkaTemplate
+import org.springframework.kafka.core.ProducerFactory
+import org.springframework.kafka.listener.ContainerProperties
+
+@Configuration
+@EnableKafka
+@Profile("dev", "prod")
+class KafkaConfig(
+    @Value("\${KAFKA_BROKERS:localhost}") private val kafkaBrokers: String,
+    @Value("\${KAFKA_TRUSTSTORE_PATH}") private val kafkaTruststorePath: String,
+    @Value("\${KAFKA_CREDSTORE_PASSWORD}") private val kafkaCredstorePassword: String,
+    @Value("\${KAFKA_KEYSTORE_PATH}") private val kafkaKeystorePath: String
+) {
+
+    @Bean
+    fun producerFactory(): ProducerFactory<String, String> {
+        return DefaultKafkaProducerFactory(producerConfigs())
+    }
+
+    @Bean
+    fun kafkaTemplate(): KafkaTemplate<String, String> {
+        return KafkaTemplate(producerFactory())
+    }
+
+    @Bean
+    fun consumerFactory(): ConsumerFactory<String, String> {
+        return DefaultKafkaConsumerFactory(consumerConfigs())
+    }
+
+    @Bean
+    fun concurrentKafkaListenerContainerFactory(kafkaErrorHandler: KafkaErrorHandler): ConcurrentKafkaListenerContainerFactory<String, String> {
+        val factory = ConcurrentKafkaListenerContainerFactory<String, String>()
+        factory.setConcurrency(1)
+        factory.containerProperties.ackMode = ContainerProperties.AckMode.MANUAL
+        factory.consumerFactory = consumerFactory()
+        factory.setCommonErrorHandler(kafkaErrorHandler)
+        return factory
+    }
```

Comment:
```suggestion
    @Bean
    fun concurrentKafkaListenerContainerFactory(kafkaErrorHandler: KafkaErrorHandler): ConcurrentKafkaListenerContainerFactory<String, String> =
        ConcurrentKafkaListenerContainerFactory<String, String>().apply {
            setConcurrency(1)
            containerProperties.ackMode = ContainerProperties.AckMode.MANUAL
            consumerFactory = consumerFactory()
            setCommonErrorHandler(kafkaErrorHandler)
        }
```
Kan bruke apply her, så får vi også gjort det om til single expression function

Reply:
Personlig er jeg litt oldschool og  ikke så glad i single expression functions, men kan godt endre sparer 1 linje :)